### PR TITLE
Initial Windows Support

### PR DESCRIPTION
**Pull request by azchohfi**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
Feature

### :arrow_heading_down: What is the current behavior?
No Windows support

### :new: What is the new behavior (if this is a feature change)?
Windows (Win32) support

### :boom: Does this PR introduce a breaking change?
No.

### :bug: Recommendations for testing
flutter run -d windows

### :memo: Links to relevant issues/docs
#788

### :thinking: Checklist before submitting

- [X] I made sure all projects build.
- [X] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy](https://dart.dev/tools/pub/versioning).
- [X] I updated CHANGELOG.md to add a description of the change.
- [X] I followed the style guide lines ([code style guide](https://github.com/Baseflow/flutter-permission-handler/blob/master/CONTRIBUTING.md)).
- [ ] I updated the relevant documentation.
- [X] I rebased onto current `master`.

### CODE REVIEW

**Comment by azchohfi** [comment]
Build failing. Waiting until #790 is fixed.

**Review by mvanbeusekom — CHANGES_REQUESTED** [review]
Hi @azchohfi,

Thanks for this contribution, really appreciate it. However we are in the process of extracting platform implementations into a separate package (as described by the Flutter [Federated architecture](https://docs.flutter.dev/development/packages-and-plugins/developing-packages#federated-plugins) recommendations). 

Therefore I would like to ask you if you could convert this implementation into a separate `permission_handler_windows` package (similar to the implementation you did for the geolocator plugin)?

**Comment by azchohfi** [comment]
Sure, no problem. I'm reverting all changes so there is no change in the main package. I'll send the second PR adding the sample app once this is merged in, and the package is on pub.dev.

File: permission_handler_windows/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: permission_handler_windows
+description: Permission plugin for Flutter. This plugin provides the Windows API to request and check permissions.
+version: 0.1.0
+homepage: https://github.com/baseflowit/flutter-permission-handler
```
Comment (mvanbeusekom) [inline permission_handler_windows/pubspec.yaml:4]: nit: I know this is not your mistake but I realised this URL is wrong. We renamed our GitHub organisation from `baseflowit` to `baseflow` several years ago and only now noticed I have never updated the homepage here. Would you mind updating this to:
```suggestion
homepage: https://github.com/baseflow/flutter-permission-handler
```

File: permission_handler_windows/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  plugin:
+    implements: permission_handler
+    platforms:
+      windows:
+        pluginClass: PermissionHandlerWindowsPlugin
+
+dependencies:
+  flutter:
+    sdk: flutter
+  meta: ^1.3.0
```
Comment (mvanbeusekom) [inline permission_handler_windows/pubspec.yaml:16]: nit: this package is not used and can be removed.

File: permission_handler_windows/pubspec.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  permission_handler_platform_interface: ^3.7.0
+
+dev_dependencies:
+  flutter_test:
+    sdk: flutter
+  mockito: ^5.0.1
+  effective_dart: ^1.3.0
```
Comment (mvanbeusekom) [inline permission_handler_windows/pubspec.yaml:23]: nit: these packages are also not used so they can also be removed.

File: permission_handler_windows/windows/permission_handler_windows_plugin.cpp [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+}
+
+}  // namespace
+
+void PermissionHandlerWindowsPluginRegisterWithRegistrar(
+    FlutterDesktopPluginRegistrarRef registrar) {
+  PermissionHandlerWindowsPlugin::RegisterWithRegistrar(
+      PluginRegistrarManager::GetInstance()
+          ->GetRegistrar<PluginRegistrarWindows>(registrar));
+}
```
Comment (mvanbeusekom) [inline permission_handler_windows/windows/permission_handler_windows_plugin.cpp:193]: nit: please include an empty line at the end of each file.
```suggestion
}

```

**Review by mvanbeusekom — CHANGES_REQUESTED** [review]
@azchohfi please also add an example App to the `permission_handler_windows` package (supporting only the Windows platform). This could be a copy of the `permission_handler_android/example` app that is slightly modified to support Windows.

I also added some minor change requests, would be great if you could go through them as well.

**Comment by azchohfi** [comment]
Done.

**Review by mvanbeusekom — APPROVED** [review]
LGTM, thank you for yet another great contribution.

**Comment by mvanbeusekom** [comment]
@azchohfi I have published this package to pub.dev. Would be great of you could create a PR which also adds Windows support to the app facing permission_handler package and update it to version 9.2.0.

If you don't have the time please let me know, I think I can also manage this.

**Comment by azchohfi** [comment]
@mvanbeusekom https://github.com/Baseflow/flutter-permission-handler/pull/797

**Pull request merged**